pallets/author-slot-filter/src/migration.rs

```diff
@@ -40,35 +41,40 @@ where
 	fn on_runtime_upgrade() -> Weight {
 		log::info!(target: "EligibleRatioToEligiblityCount", "starting migration");
 
-		if let Some(old_value) =
-			migration::get_storage_value::<Percent>(PALLET_NAME, ELIGIBLE_RATIO_ITEM_NAME, &[])
-		{
-			let total_authors = <T as Config>::PotentialAuthors::get().len();
-			let new_value: u32 = percent_of_num(old_value, total_authors as u32);
-			migration::put_storage_value(
-				PALLET_NAME,
-				ELIGIBLE_COUNT_ITEM_NAME,
-				&[],
-				NonZeroU32::new(new_value).unwrap_or(crate::pallet::DEFAULT_TOTAL_ELIGIBLE_AUTHORS),
-			);
-
-			let db_weights = T::DbWeight::get();
-			db_weights.write + db_weights.read
-		} else {
-			0
-		}
+		let old_value =
+			migration::get_storage_value::<Percent>(PALLET_NAME, ELIGIBLE_RATIO_ITEM_NAME, &[]);
+		let new_value = old_value
+			.map(|value| {
+				let total_authors = <T as Config>::PotentialAuthors::get().len();
+				let new_value = percent_of_num(value, total_authors as u32);
+				NonZeroU32::new(new_value)
+			})
+			.flatten()
```

Comment:
You should not use flatten here, maybe replace the whole block code with an if let syntax